src/legacy/utils/price.ts

```diff
@@ -20,7 +23,20 @@ export type QuoteResult = [PromiseSettledResult<PriceInformation>, PromiseSettle
  * Queries the new Quote api endpoint found here: https://protocol-mainnet.gnosis.io/api/#/default/post_api_v1_quote
  * TODO: consider name // check with backend when logic returns best quote, not first
  */
-export async function getFullQuote({ quoteParams }: { quoteParams: LegacyFeeQuoteParams }): Promise<QuoteResult> {
+export async function getFullQuote({
+  quoteParams,
+  permitHookParams,
+}: {
+  quoteParams: LegacyFeeQuoteParams
+  permitHookParams?: PermitHookParams
+}): Promise<QuoteResult> {
+  // TODO
+  const isTokenSupportsEIP2612 = true
+
+  if (isTokenSupportsEIP2612 && permitHookParams) {
+    ;(quoteParams as any).appData = toKeccak256(await generatePermitHook(permitHookParams))
```

Comment:
The current implementation of `generatePermitHook` requires signing a typed data via user's provider. However, to get a quote it doesn't matter which wallet is signed the permit. So, we can use arbitrary private key and avoid interaction with user's wallet.

Reply:
Ahhhh interesting!
That's great to know, will work on it.

Reply:
Yes

Reply:
Just before signing an order